Failed to transform Message when switch header-enricher from 1.3 to 2.x in Spring Cloud Dataflow

Hi,

I create the Spring Cloud Data Flow stream with header-enricher 1.3 and deploy it , it works fine. But when I switch to header-enricher 2.x in same SCDF environment, it throw "Failed to transform Message exception".

I'm wondering what's the difference between 1.3 and 2.x that cause the issue?

**Here is 1.3 stream definition:**

```
app register --name header-enricher_1_3 --type processor --uri http://repo.spring.io/release/org/springframework/cloud/stream/app/header-enricher-processor-rabbit/1.3.1.RELEASE/header-enricher-processor-rabbit-1.3.1.RELEASE.jar 

stream create Header_TEST --definition "trigger --time-unit=MINUTES --initialDelay=0  --fixed-delay=500  --payload={'testWords':'Hello World!'} | header-enricher_1_3  --headers='testWords_header=payload.testWords' --outputType=application/json  | log --expression=headers.testWords_header" --deploy
```
 
It works fine, we can get " Hello World!" in log:

```
2019-04-04T17:15:08.833-04:00 [APP/PROC/WEB/0] [OUT] 2019-04-04 21:15:08.831 INFO 17 --- [ main] o.s.c.s.a.l.s.r.LogSinkRabbitApplication : Started LogSinkRabbitApplication in 12.406 seconds (JVM running for 14.73)
2019-04-04T17:15:10.312-04:00 [CELL/0] [OUT] Container became healthy
2019-04-04T17:15:13.112-04:00 [APP/PROC/WEB/0] [OUT] 2019-04-04 21:15:13.111 INFO 17 --- [w.Header_TEST-9] dataflow-server-stg-Header_TEST-log : Hello World!
```

**Here is the 2.x Stream definition:**

```
app register --name header-enricher_2_1 --type processor --uri	http://repo.spring.io/release/org/springframework/cloud/stream/app/header-enricher-processor-rabbit/2.1.0.RELEASE/header-enricher-processor-rabbit-2.1.0.RELEASE.jar 

stream create Header_TEST --definition "trigger --time-unit=MINUTES --initialDelay=0  --fixed-delay=500  --payload={'testWords':'Hello World!'} | header-enricher_2_1  --headers='testWords_header=payload.testWords' --outputType=application/json  | log --expression=headers.testWords_header" --deploy
```

I got an exception says:  
_**org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]'**_

### Here is the completed log:

```
019-04-04T17:19:38.425-04:00 [APP/PROC/WEB/0] [OUT] 2019-04-04 21:19:38.423 INFO 16 --- [ main] HeaderEnricherProcessorRabbitApplication : Started HeaderEnricherProcessorRabbitApplication in 12.71 seconds (JVM running for 13.979)
2019-04-04T17:19:38.802-04:00 [CELL/0] [OUT] Container became healthy
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] ... 31 more
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:128)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:169)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:371)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.SpelNodeImpl.getTypedValue(SpelNodeImpl.java:116)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:89)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference$AccessorLValue.getValue(PropertyOrFieldReference.java:407)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.access$000(PropertyOrFieldReference.java:51)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.getValueInternal(PropertyOrFieldReference.java:104)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.readProperty(PropertyOrFieldReference.java:217)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] ... 28 more
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.HeaderEnricher.transform(HeaderEnricher.java:122)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.support.ExpressionEvaluatingHeaderValueMessageProcessor.processMessage(ExpressionEvaluatingHeaderValueMessageProcessor.java:71)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.ExpressionEvaluatingMessageProcessor.processMessage(ExpressionEvaluatingMessageProcessor.java:105)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:136)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?, failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}]
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] ... 27 more
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.MessageTransformingHandler.handleRequestMessage(MessageTransformingHandler.java:89)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.HeaderEnricher.transform(HeaderEnricher.java:131)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.messaging.MessagingException: failed to transform message headers; nested exception is org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?, failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}], failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}]
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at java.lang.Thread.run(Thread.java:748)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$AsyncMessageProcessingConsumer.run(SimpleMessageListenerContainer.java:1042)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.access$700(SimpleMessageListenerContainer.java:77)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.receiveAndExecute(SimpleMessageListenerContainer.java:801)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.doReceiveAndExecute(SimpleMessageListenerContainer.java:817)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.executeListener(AbstractMessageListenerContainer.java:1303)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:1324)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.actualInvokeListener(AbstractMessageListenerContainer.java:1337)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:1414)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter$Listener.onMessage(AmqpInboundChannelAdapter.java:211)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:180)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:287)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter$Listener.lambda$onMessage$0(AmqpInboundChannelAdapter.java:214)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter.access$1100(AmqpInboundChannelAdapter.java:60)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.endpoint.MessageProducerSupport.sendMessage(MessageProducerSupport.java:203)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:108)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:160)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:181)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:394)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:445)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:73)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:105)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:132)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.AbstractDispatcher.tryOptimizedDispatch(AbstractDispatcher.java:116)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:158)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:109)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.MessageTransformingHandler.handleRequestMessage(MessageTransformingHandler.java:95)
2019-04-04T17:19:43.338-04:00 [APP/PROC/WEB/0] [OUT] 2019-04-04 21:19:43.337 ERROR 16 --- [r.Header_TEST-9] o.s.integration.handler.LoggingHandler : org.springframework.integration.transformer.MessageTransformationException: Failed to transform Message; nested exception is org.springframework.messaging.MessagingException: failed to transform message headers; nested exception is org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?, failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}], failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}], failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}]
2019-04-04T17:19:43.345-04:00 [APP/PROC/WEB/0] [OUT] 2019-04-04 21:19:43.345 WARN 16 --- [r.Header_TEST-9] o.s.a.r.r.RejectAndDontRequeueRecoverer : Retries exhausted for message (Body:'{"testWords":"Hello World!"}' MessageProperties [headers={contentType=application/json}, contentType=application/json, contentLength=0, receivedDeliveryMode=PERSISTENT, priority=0, redelivered=false, receivedExchange=Header_TEST.trigger, receivedRoutingKey=Header_TEST.trigger, deliveryTag=1, consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, consumerQueue=Header_TEST.trigger.Header_TEST])
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] ... 31 common frames omitted
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:128) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:169) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:371) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.SpelNodeImpl.getTypedValue(SpelNodeImpl.java:116) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:89) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference$AccessorLValue.getValue(PropertyOrFieldReference.java:407) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.access$000(PropertyOrFieldReference.java:51) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.getValueInternal(PropertyOrFieldReference.java:104) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.readProperty(PropertyOrFieldReference.java:217) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] ... 28 common frames omitted
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.HeaderEnricher.transform(HeaderEnricher.java:122) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.support.ExpressionEvaluatingHeaderValueMessageProcessor.processMessage(ExpressionEvaluatingHeaderValueMessageProcessor.java:71) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.ExpressionEvaluatingMessageProcessor.processMessage(ExpressionEvaluatingMessageProcessor.java:105) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:136) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] ... 27 common frames omitted
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.MessageTransformingHandler.handleRequestMessage(MessageTransformingHandler.java:89) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.HeaderEnricher.transform(HeaderEnricher.java:131) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.messaging.MessagingException: failed to transform message headers; nested exception is org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?, failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_192]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$AsyncMessageProcessingConsumer.run(SimpleMessageListenerContainer.java:1042) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.access$700(SimpleMessageListenerContainer.java:77) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.receiveAndExecute(SimpleMessageListenerContainer.java:801) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.doReceiveAndExecute(SimpleMessageListenerContainer.java:817) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.executeListener(AbstractMessageListenerContainer.java:1303) [spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:1324) [spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.actualInvokeListener(AbstractMessageListenerContainer.java:1337) [spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:1414) [spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter$Listener.onMessage(AmqpInboundChannelAdapter.java:211) [spring-integration-amqp-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:180) [spring-retry-1.2.2.RELEASE.jar!/:na]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:287) [spring-retry-1.2.2.RELEASE.jar!/:na]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter$Listener.lambda$onMessage$0(AmqpInboundChannelAdapter.java:214) [spring-integration-amqp-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter.access$1100(AmqpInboundChannelAdapter.java:60) ~[spring-integration-amqp-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.endpoint.MessageProducerSupport.sendMessage(MessageProducerSupport.java:203) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:108) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:160) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:181) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:394) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:445) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:73) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:105) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:132) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.AbstractDispatcher.tryOptimizedDispatch(AbstractDispatcher.java:116) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:158) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:109) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.MessageTransformingHandler.handleRequestMessage(MessageTransformingHandler.java:95) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.346-04:00 [APP/PROC/WEB/0] [OUT] org.springframework.integration.transformer.MessageTransformationException: Failed to transform Message; nested exception is org.springframework.messaging.MessagingException: failed to transform message headers; nested exception is org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?, failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}], failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.support.ErrorMessagePublisher.publish(ErrorMessagePublisher.java:164) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:98) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:108) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:160) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:181) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:394) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:475) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.support.utils.IntegrationUtils.wrapInDeliveryExceptionIfNecessary(IntegrationUtils.java:163) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.messaging.MessageDeliveryException: failed to send Message to channel 'Header_TEST.trigger.Header_TEST.errors'; nested exception is org.springframework.amqp.rabbit.listener.exception.ListenerExecutionFailedException: Retry Policy Exhausted
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at java.lang.Thread.run(Thread.java:748) [na:1.8.0_192]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$AsyncMessageProcessingConsumer.run(SimpleMessageListenerContainer.java:1042) [spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.access$700(SimpleMessageListenerContainer.java:77) [spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.receiveAndExecute(SimpleMessageListenerContainer.java:801) [spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.doReceiveAndExecute(SimpleMessageListenerContainer.java:817) [spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.executeListener(AbstractMessageListenerContainer.java:1303) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:1324) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.actualInvokeListener(AbstractMessageListenerContainer.java:1337) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:1417) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.wrapToListenerExecutionFailedExceptionIfNeeded(AbstractMessageListenerContainer.java:1506) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] org.springframework.amqp.rabbit.listener.exception.ListenerExecutionFailedException: Listener threw exception
2019-04-04T17:19:43.348-04:00 [APP/PROC/WEB/0] [OUT] 2019-04-04 21:19:43.348 WARN 16 --- [r.Header_TEST-9] s.a.r.l.ConditionalRejectingErrorHandler : Execution of Rabbit message listener failed.
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] ... 31 common frames omitted
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:128) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:169) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:371) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.SpelNodeImpl.getTypedValue(SpelNodeImpl.java:116) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:89) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference$AccessorLValue.getValue(PropertyOrFieldReference.java:407) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.access$000(PropertyOrFieldReference.java:51) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.getValueInternal(PropertyOrFieldReference.java:104) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.expression.spel.ast.PropertyOrFieldReference.readProperty(PropertyOrFieldReference.java:217) ~[spring-expression-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] ... 28 common frames omitted
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.HeaderEnricher.transform(HeaderEnricher.java:122) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.support.ExpressionEvaluatingHeaderValueMessageProcessor.processMessage(ExpressionEvaluatingHeaderValueMessageProcessor.java:71) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.ExpressionEvaluatingMessageProcessor.processMessage(ExpressionEvaluatingMessageProcessor.java:105) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.util.AbstractExpressionEvaluator.evaluateExpression(AbstractExpressionEvaluator.java:136) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] ... 27 common frames omitted
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.MessageTransformingHandler.handleRequestMessage(MessageTransformingHandler.java:89) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.HeaderEnricher.transform(HeaderEnricher.java:131) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.messaging.MessagingException: failed to transform message headers; nested exception is org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?, failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] ... 11 common frames omitted
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:287) ~[spring-retry-1.2.2.RELEASE.jar!/:na]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter$Listener.lambda$onMessage$0(AmqpInboundChannelAdapter.java:214) ~[spring-integration-amqp-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter.access$1100(AmqpInboundChannelAdapter.java:60) ~[spring-integration-amqp-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.endpoint.MessageProducerSupport.sendMessage(MessageProducerSupport.java:203) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:108) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:160) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:181) ~[spring-messaging-5.0.7.RELEASE.jar!/:5.0.7.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:394) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:445) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:73) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:105) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:132) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.AbstractDispatcher.tryOptimizedDispatch(AbstractDispatcher.java:116) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:158) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:109) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.transformer.MessageTransformingHandler.handleRequestMessage(MessageTransformingHandler.java:95) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.integration.transformer.MessageTransformationException: Failed to transform Message; nested exception is org.springframework.messaging.MessagingException: failed to transform message headers; nested exception is org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?, failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}], failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.amqp.AmqpRejectAndDontRequeueException: org.springframework.integration.transformer.MessageTransformationException: Failed to transform Message; nested exception is org.springframework.messaging.MessagingException: failed to transform message headers; nested exception is org.springframework.messaging.MessageHandlingException: Expression evaluation failed: payload.testWords; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1008E: Property or field 'testWords' cannot be found on object of type 'byte[]' - maybe not public or not valid?, failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}], failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}], failedMessage=GenericMessage [payload=byte[28], headers={amqp_receivedDeliveryMode=PERSISTENT, amqp_receivedRoutingKey=Header_TEST.trigger, amqp_receivedExchange=Header_TEST.trigger, amqp_deliveryTag=1, deliveryAttempt=3, amqp_consumerQueue=Header_TEST.trigger.Header_TEST, amqp_redelivered=false, id=66e0aa1c-1188-5c0b-ed94-cdfea662e096, amqp_consumerTag=amq.ctag-I7THOoYQhfPXXp5pivECXg, contentType=application/json, timestamp=1554412780307}]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] ... 21 common frames omitted
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.BroadcastingDispatcher.dispatch(BroadcastingDispatcher.java:180) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.dispatcher.BroadcastingDispatcher.invokeHandler(BroadcastingDispatcher.java:224) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.cloud.stream.binder.rabbit.RabbitMessageChannelBinder$3.handleMessage(RabbitMessageChannelBinder.java:513) ~[spring-cloud-stream-binder-rabbit-2.0.1.RELEASE.jar!/:2.0.1.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.retry.RejectAndDontRequeueRecoverer.recover(RejectAndDontRequeueRecoverer.java:45) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] Caused by: org.springframework.amqp.rabbit.listener.exception.ListenerExecutionFailedException: Retry Policy Exhausted
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] ... 8 common frames omitted
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:1414) ~[spring-rabbit-2.0.4.RELEASE.jar!/:2.0.4.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter$Listener.onMessage(AmqpInboundChannelAdapter.java:211) ~[spring-integration-amqp-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:180) ~[spring-retry-1.2.2.RELEASE.jar!/:na]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:351) ~[spring-retry-1.2.2.RELEASE.jar!/:na]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.retry.support.RetryTemplate.handleRetryExhausted(RetryTemplate.java:512) ~[spring-retry-1.2.2.RELEASE.jar!/:na]
2019-04-04T17:19:43.349-04:00 [APP/PROC/WEB/0] [OUT] at org.springframework.integration.handler.advice.ErrorMessageSendingRecoverer.recover(ErrorMessageSendingRecoverer.java:83) ~[spring-integration-core-5.0.6.RELEASE.jar!/:5.0.6.RELEASE]


```

--- a/spring-cloud-starter-stream-processor-header-enricher/src/main/java/org/springframework/cloud/stream/app/header/enricher/processor/HeaderEnricherProcessorConfiguration.java
+++ b/spring-cloud-starter-stream-processor-header-enricher/src/main/java/org/springframework/cloud/stream/app/header/enricher/processor/HeaderEnricherProcessorConfiguration.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017 the original author or authors.
+ * Copyright 2017-2019 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -25,17 +25,22 @@ import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.config.ConfigurableBeanFactory;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.config.BindingProperties;
 import org.springframework.cloud.stream.messaging.Processor;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Scope;
 import org.springframework.integration.annotation.Transformer;
+import org.springframework.integration.support.MutableMessage;
 import org.springframework.integration.transformer.HeaderEnricher;
 import org.springframework.integration.transformer.support.ExpressionEvaluatingHeaderValueMessageProcessor;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
 
 /**
  * A Processor app that adds expression-evaluated headers.
  *
  * @author Gary Russell
+ * @author Christian Tzolov
  */
 @EnableBinding(Processor.class)
 @EnableConfigurationProperties(HeaderEnricherProcessorProperties.class)
@@ -46,7 +51,7 @@ public class HeaderEnricherProcessorConfiguration {
 
 	@Bean
 	@Transformer(inputChannel = Processor.INPUT, outputChannel = Processor.OUTPUT)
-	public HeaderEnricher headerEnricher() throws Exception {
+	public HeaderEnricher headerEnricher() {
 		Map<String, ExpressionEvaluatingHeaderValueMessageProcessor<?>> headersToAdd = new HashMap<>();
 		Properties props = this.properties.getHeaders();
 		Enumeration<?> enumeration = props.propertyNames();
@@ -54,7 +59,20 @@ public class HeaderEnricherProcessorConfiguration {
 			String propertyName = (String) enumeration.nextElement();
 			headersToAdd.put(propertyName, processor(props.getProperty(propertyName)));
 		}
-		HeaderEnricher headerEnricher = new HeaderEnricher(headersToAdd);
+		HeaderEnricher headerEnricher = new HeaderEnricher(headersToAdd) {
+			@Override
+			public Message<?> transform(Message<?> message) {
+				if (message.getPayload() instanceof byte[]){
+					String contentType = message.getHeaders().containsKey(MessageHeaders.CONTENT_TYPE)
+							? message.getHeaders().get(MessageHeaders.CONTENT_TYPE).toString()
+							: BindingProperties.DEFAULT_CONTENT_TYPE.toString();
+					if (contentType.contains("text") || contentType.contains("json") || contentType.contains("x-spring-tuple")) {
+						message = new MutableMessage<>(new String(((byte[]) message.getPayload())), message.getHeaders());
+					}
+				}
+				return super.transform(message);
+			}
+		};
 		headerEnricher.setDefaultOverwrite(this.properties.isOverwrite());
 		return headerEnricher;
 	}

--- a/spring-cloud-starter-stream-processor-header-enricher/src/test/java/org/springframework/cloud/stream/app/header/enricher/processor/HeaderEnricherProcessorIntegrationTests.java
+++ b/spring-cloud-starter-stream-processor-header-enricher/src/test/java/org/springframework/cloud/stream/app/header/enricher/processor/HeaderEnricherProcessorIntegrationTests.java
@@ -56,7 +56,8 @@ public abstract class HeaderEnricherProcessorIntegrationTests {
 	protected MessageCollector collector;
 
 	protected void doTest() throws InterruptedException {
-		this.channels.input().send(MessageBuilder.withPayload("hello").setHeader("baz", "qux").build());
+		this.channels.input().send(MessageBuilder.withPayload("hello".getBytes())
+				.setHeader("baz", "qux").build());
 		Message<?> out = this.collector.forChannel(this.channels.output()).poll(10, TimeUnit.SECONDS);
 		assertThat(out, notNullValue());
 		assertThat(out, HeaderMatcher.hasHeader("foo", equalTo("bar")));

